1023570 - libnetsnmpagent.so crash under certain conditions

commit 94a499c5b81e4d9fa01f78719bc578c997520151
Author: Dave Shield <dts12@users.sourceforge.net>
Date:   Tue Dec 1 11:56:10 2009 +0000

    CHANGES: agent: BUGS: 2841008: Protect against crash following failed AgentX response
    
    
    git-svn-id: file:///home/hardaker/lib/sf-bkups/net-snmp-convert-svnrepo/trunk@17856 06827809-a52a-0410-b366-d66718629ded

diff --git a/agent/mibgroup/agentx/client.c b/agent/mibgroup/agentx/client.c
index 0d9ddb6..624cad4 100644
--- a/agent/mibgroup/agentx/client.c
+++ b/agent/mibgroup/agentx/client.c
@@ -65,7 +65,7 @@ agentx_synch_input(int op,
     struct synch_state *state = (struct synch_state *) magic;
     struct timeval  now, diff;
 
-    if (reqid != state->reqid) {
+    if (!state || reqid != state->reqid) {
         return handle_agentx_packet(op, session, reqid, pdu, magic);
     }
 
@@ -139,6 +139,9 @@ agentx_open_session(netsnmp_session * ss)
     if (agentx_synch_response(ss, pdu, &response) != STAT_SUCCESS)
         return 0;
 
+    if (!response)
+        return 0;
+
     if (response->errstat != SNMP_ERR_NOERROR) {
         snmp_free_pdu(response);
         return 0;
